alacritty_terminal/src/term/mod.rs

```diff
@@ -999,6 +1003,8 @@ impl<T> Term<T> {
         self.grid.resize(!is_alt, num_lines, num_cols);
         self.inactive_grid.resize(is_alt, num_lines, num_cols);
 
+        self.size = size;
```

Comment:
This should be at the top, even before the num_cols/num_lines change. It's possible that the terminal resized but the number of columns/lines stays the same.